bpfd-operator/controllers/bpfd-operator/common.go

```diff
@@ -188,3 +190,50 @@ func statusChangedPredicate() predicate.Funcs {
 		},
 	}
 }
+
+func (r *ReconcilerCommon) updateCondition(ctx context.Context, obj client.Object, conditions *[]metav1.Condition, cond bpfdiov1alpha1.ProgramConditionType, message string) (ctrl.Result, error) {
```

Comment:
Can you move this and the one in bpfd-agent/common.go into `/bpfd-operator/internal/k8s.go` ?